venvs/tests/test_make.py

```diff
@@ -140,6 +140,44 @@ def test_multiple_installs_one_link(self):
     def test_multiple_installs_one_link_no_name(self):
         self.run_cli(["-i", "foo", "-i", "bar", "-l", "foo"], exit_status=2)
 
+    def test_install_edit_config(self):
+        """Install will automatically edit the config file."""
+        self.run_cli(["-l", "foo", "-i", "bar", "--persist"])
+        contents = _load_config(
+            filesystem=self.filesystem,
+            locator=self.locator,
+        )
+        self.assertEqual(
+            contents,
+            {'virtualenv': {"bar": {"install": ["bar"], "link": ["foo"]}}}
+        )
+
+    def test_handle_empty_config_file(self):
+        """Don't'break with an empty config file."""
+
+        # Empty the config file.
+        with self.filesystem.open(
+                self.locator.root.descendant("virtualenvs.toml"),
+                "w",
+        ):
+            pass
+
+        self.run_cli(["-l", "foo", "-i", "bar", "--persist"])
+        contents = _load_config(
+            filesystem=self.filesystem,
+            locator=self.locator,
+        )
+        self.assertEqual(
+            contents,
+            {'virtualenv': {"bar": {"install": ["bar"], "link": ["foo"]}}}
+        )
+
+    def test_install_no_persist(self):
+        """Install --no-persist will not edit the config file."""
+        self.run_cli(["-l", "foo", "-i", "bar", "--no-persist"])
+        with self.assertRaises(filesystems.exceptions.FileNotFound):
+            _load_config(filesystem=self.filesystem, locator=self.locator)
+
```

Comment:
Think there's one more test case missing (with behavior) for the case where the directory itself does not exist in which you'd place the config file.